File: worksheets/dp-four/scala/src/visitor/HtmlParentElement.scala
package visitor

case class HtmlParentElement(var tagName: String) extends HtmlTag {
  var startTag: String = ""
  var endTag: String = ""

  private var childrenTag = ???

  override def addChildTag(htmlTag: HtmlTag): Unit = ???
  override def removeChildTag(htmlTag: HtmlTag): Unit = ???
  override def getChildren() = childrenTag
  override def generateHtml(): Unit = ???
  override def accept(visitor: Visitor): Unit = ???

  def getEndTag(): String = ???
  def getStartTag(): String = ???
  def getTagName(): String = ???
  def setEndTag(tag: String): Unit = ???
  def setStartTag(tag: String): Unit = ???
}

File: worksheets/dp-four/kotlin/src/visitor/HtmlParentElement.kt
package visitor

data class HtmlParentElement(val startTag: String,
                             val endTag: String) : HtmlTag(startTag, endTag) {
    private var childrenTag: List<HtmlTag> = TODO()

    override fun addChildTag(htmlTag: HtmlTag) {
        TODO()
    }

    override fun removeChildTag(htmlTag: HtmlTag) {
        TODO()
    }

    override fun getChildren() = childrenTag

    override fun generateHtml() {
        TODO()
    }

    override fun accept(visitor: Visitor) {
        TODO()
    }
}
